lib/libxdp/xsk.c

```diff
@@ -735,6 +856,9 @@ static int __xsk_setup_xdp_prog(struct xsk_socket *xsk, int *xsks_map_fd)
 		xdp_program__detach(ctx->xdp_prog, ctx->ifindex,
 				    xsk_convert_xdp_flags(xsk->config.xdp_flags), 0);
 err_prog_load:
+	if (ctx->refcnt_map_fd)
+		close(ctx->refcnt_map_fd);
+	ctx->refcnt_map_fd = 0;
```

Comment:
Do we generally guard against `0` being a valid fd?

Reply:
Right, but in libxdp.c we have a separate `prog_fd = fcntl(fd, F_DUPFD_CLOEXEC, MIN_FD);` to make sure the fd is never 0. But seems libbpf actually does that for any fd it returns (since 0.6.0), so we should be fine in any case...